src/components/task-plan/homework.cjsx

```diff
@@ -0,0 +1,404 @@
+  mixins: [ExerciseCardMixin]
+
+  moveExerciseUp: ->
+    TaskPlanActions.moveExercise(@props.planId, @props.exercise, -1)
+
+  moveExerciseDown: ->
+    TaskPlanActions.moveExercise(@props.planId, @props.exercise, 1)
+
+  removeExercise: ->
+    TaskPlanActions.removeExercise(@props.planId, @props.exercise)
+
+  renderHeader: ->
+    <span>
+      <span class="-exercise-number">{@props.index+1}</span>
+      <BS.Button bsStyle="info" onClick={@moveExerciseUp} className="-move-exercise-up">^</BS.Button>
+      <BS.Button bsStyle="info" onClick={@moveExerciseDown} className="-move-exercise-up">v</BS.Button>
+      <BS.Button bsStyle="danger" onClick={@removeExercise} className="-remove-exercise-up">X</BS.Button>
+    </span>
+
+  getPanelStyle: ->
+    "default"
+
+  render: ->
+    @renderExercise()
+
+
+AddExerciseCard = React.createClass
+  mixins: [ExerciseCardMixin]
+
+  toggleExercise: ->
+    if TaskPlanStore.hasExercise(@props.planId, @props.exercise.id)
+      TaskPlanActions.removeExercise(@props.planId, @props.exercise)
+    else
+      TaskPlanActions.addExercise(@props.planId, @props.exercise)
+
+  renderHeader: ->
+    active = TaskPlanStore.hasExercise(@props.planId, @props.exercise.id)
+    toggleText = if not active then <span>+</span> else <span>-</span>
+    <span><BS.Button bsStyle="primary" onClick={@toggleExercise} className="-add-exercise">{toggleText}</BS.Button></span>
+
+  getPanelStyle: ->
+    if TaskPlanStore.hasExercise(@props.planId, @props.exercise.id)
+      return "info"
+    else
+      return "default"
+
+  render: ->
+    @renderExercise()
+
+ExercisesRenderMixin =
+  componentWillMount:   -> ExerciseStore.addChangeListener(@update)
+  componentWillUnmount: -> ExerciseStore.removeChangeListener(@update)
+
+  update: ->
+    @setState({})
+
+
+  renderLoading: ->
+    {courseId, pageIds, hide} = @props
+
+    unless @props.shouldShow
+      return <span className="-no-show"/>
+    unless ExerciseStore.isLoaded(pageIds)
+      ExerciseActions.load(courseId, pageIds)
+      return <span className="-loading">Loading...</span>
+    
+    false
+
+
+ReviewExercises = React.createClass
+  mixins: [ExercisesRenderMixin]
+
+  renderExercise: (exercise, i) ->
+    <ReviewExerciseCard index={i} planId={@props.planId} exercise={exercise}/>
+
+  render: ->
+    load = @renderLoading()
+    if (load)
+      return load
+    {courseId, pageIds, hide, planId} = @props
+    exercise_ids = TaskPlanStore.getExercises(planId)
+    exercises = _.map(exercise_ids, ExerciseStore.getExerciseById)
+    renderedExercises = _.map(exercises, @renderExercise)
+    
+    <div>
+      {renderedExercises}
+    </div>
+
+AddExercises = React.createClass
+  mixins: [ExercisesRenderMixin]
+
+  renderExercise: (exercise) ->
+    <AddExerciseCard planId={@props.planId} exercise={exercise}/>
+
+  renderInRows: (renderedExercises) ->
+    rows = []
+    i=0
+    while i < renderedExercises.length
+      left = renderedExercises[i]
```

Comment:
I did it this way because sometimes the homework cards are variable heights, so that it would break the layout if the left card was really long.